test/test_execute.py

```diff
@@ -1863,131 +1714,134 @@ def test_output_from(self):
-print(f'Workflow {workflow_id}: master id={master_id}')
-assert step_name == 'nested'
-assert workflow_id != master_id
-
-[default]
-print(f'Workflow {workflow_id}: step name={step_name}')
-print(f'Workflow {workflow_id}: step id={step_id}')
-print(f'Workflow {workflow_id}: workflow id={workflow_id}')
-print(f'Workflow {workflow_id}: master id={master_id}')
-assert step_name == 'default'
-assert workflow_id == master_id
-sos_run('nested')
-''')
-        wf = script.workflow()
-        Base_Executor(wf).run()
+        execute_workflow('''
+            [nested]
+            print(f'Workflow {workflow_id}: step name={step_name}')
+            print(f'Workflow {workflow_id}: step id={step_id}')
+            print(f'Workflow {workflow_id}: workflow id={workflow_id}')
+            print(f'Workflow {workflow_id}: master id={master_id}')
+            assert step_name == 'nested'
+            assert workflow_id != master_id
+
+            [default]
+            print(f'Workflow {workflow_id}: step name={step_name}')
+            print(f'Workflow {workflow_id}: step id={step_id}')
+            print(f'Workflow {workflow_id}: workflow id={workflow_id}')
+            print(f'Workflow {workflow_id}: master id={master_id}')
+            assert step_name == 'default'
+            assert workflow_id == master_id
+            sos_run('nested')
+            ''')
 
     def test_reexecution_of_dynamic_depends(self):
         '''Testing the rerun of steps to verify dependency'''
         for file in ('a.bam', 'a.bam.bai'):
             if os.path.isfile(file):
                 os.remove(file)
-        script = SoS_Script('''
-[BAI: provides='{filename}.bam.bai']
-_output.touch()
+        script = SoS_Script(textwrap.dedent('''
+        [BAI: provides='{filename}.bam.bai']
+        _output.touch()
 
-[BAM]
-output: 'a.bam'
-_output.touch()
+        [BAM]
+        output: 'a.bam'
+        _output.touch()
 
-[default]
-input: 'a.bam'
-depends: _input.with_suffix('.bam.bai')
-''')
+        [default]
+        input: 'a.bam'
+        depends: _input.with_suffix('.bam.bai')
+        '''))
         wf = script.workflow()
         Base_Executor(wf).run()
         # if we run again, because depends, the step will be re-checked
         os.remove('a.bam')
         res = Base_Executor(wf).run()
-        self.assertEqual(res['__completed__']['__step_completed__'], 2)
-        self.assertEqual(res['__completed__']['__step_skipped__'], 0)
+        assert res['__completed__']['__step_completed__'] == 2
+        assert res['__completed__']['__step_skipped__'] == 0
         #
         os.remove('a.bam')
         res = Base_Executor(wf, config={'trace_existing': True}).run()
-        self.assertEqual(res['__completed__']['__step_completed__'], 2)
-        self.assertEqual(res['__completed__']['__step_skipped__'], 1)
+        assert res['__completed__']['__step_completed__'] == 2
+        assert res['__completed__']['__step_skipped__'] == 1
```

Comment:
I tried to make the script variable a string and change the line 
`res = Base_Executor(wf, config={'trace_existing': True}).run()`
to
`res = execute_workflow(script,config={'trace_existing': True})`

But this resulted in a failed assertion statement. Is my syntax incorrect?

Reply:
I think in `execute_workflow` we use `options` for such things and `config` for replacing user-specified config file.

Reply:
https://github.com/vatlab/sos/blob/master/src/sos/__init__.py#L47